aggregator/src/app/AggregationStrategy.ts

```diff
@@ -231,22 +236,34 @@ export default class AggregationStrategy {
       assert(after.success);
 
       const bundleResult = processBundleResults[i];
-
-      let success: boolean;
-
-      if (bundleResult.success) {
-        const [operationResults] = bundleResult.returnValue;
-
-        // We require that at least one operation succeeds, even though
-        // processBundle doesn't revert in this case.
-        success = operationResults.some((opSuccess) => opSuccess === true);
-      } else {
-        success = false;
+      const fee = after.returnValue[0].sub(before.returnValue[0]);
+      if (!bundleResult.success) {
+        return { success: false, fee, errorReason: null };
       }
 
-      const fee = after.returnValue[0].sub(before.returnValue[0]);
+      const [operationStatuses, results] = bundleResult.returnValue;
+
+      let errorReason: OperationResultError | null = null;
+      // We require that at least one operation succeeds, even though
+      // processBundle doesn't revert in this case.
+      const success = operationStatuses.some((opSuccess: boolean) => opSuccess === true);
+      if (!success) {
+        const error = results.map((result: string[]) => {
+          try {
+            if (result[0]) {
+              return decodeError(result[0]);
+            }
+            return;
+          } catch (_err) {
+            return;
+          }
+        });
+        errorReason = error[0] !== undefined
+          ? error[0]
+          : null;
+      }
 
-      return { success, fee };
+      return { success, fee, errorReason };
```

Comment:
Returning the error reason here so that we can link it to a failed row later when we check the index of the failed bundle